src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs

```diff
@@ -1089,19 +1010,88 @@ protected override Size ArrangeOverride(Size finalSize)
 				var padding = Padding;
 				var borderThickness = BorderThickness;
 
-				var finalRect = new Windows.Foundation.Rect(
+				var innerRect = new Windows.Foundation.Rect(
 					padding.Left + borderThickness.Left,
 					padding.Top + borderThickness.Top,
 					finalSize.Width - padding.Left - padding.Right - borderThickness.Left - borderThickness.Right,
 					finalSize.Height - padding.Top - padding.Bottom - borderThickness.Top - borderThickness.Bottom
 				);
 
-				base.ArrangeElement(child, finalRect);
+				var availableSize = new Size(innerRect.Width, innerRect.Height);
+
+				// Using GetElementDesiredSize to properly handle native controls.
+				var desiredSize = GetElementDesiredSize(child);
+
+				var contentWidth = HorizontalContentAlignment == HorizontalAlignment.Stretch ?
```

Comment:
the generic implementation of arranging given the content and available size can be move to LayoutHelper
it is something that is used frequently, yet the implementation is never reused

Reply:
Will refactor and move it there

Reply:
In the end I decided against moving it, I can explain over Teams :-) .